CMakeLists.txt

```diff
@@ -1,227 +1,182 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.20)
 project(
 	zenohc
 	VERSION 0.7.0.1
```

Comment:
We need autoversion! Smth like this: https://github.com/nocnokneo/cmake-git-versioning-example

Reply:
I agree, but this is out of scope of this update. Created issue for this https://github.com/eclipse-zenoh/zenoh-c/issues/142